library/core/src/num/nonzero.rs

```diff
@@ -1110,6 +1117,12 @@ macro_rules! nonzero_integer {
             #[must_use = "this returns the result of the operation, \
                           without modifying the original"]
             #[inline]
+            #[requires({
+                self.get().checked_mul(other.get()).is_some()
+            })]
+            #[ensures(|result: &Self| {
+                self.get().checked_mul(other.get()).unwrap() == result.get()
```

Comment:
```suggestion
                self.get().checked_mul(other.get()).is_some_and(|product| product == result.get())
```
Best to avoid unwrapping when possible IMO, even if we know from the precondition that it's fine.